src/features/NoteContext.jsx

```diff
@@ -0,0 +1,69 @@
+import { createContext } from "react";
+import { useState } from "react";
+
+export const TodoContext = createContext([]);
+
+export const TodoContextProvider = ({ children }) => {
+  const [todos, setTodos] = useState([]);
+  const [newTodo, setNewTodo] = useState("");
+  const [isEditing, setIsEditing] = useState({});
+
+  const handleNewTodoChange = (event) => {
+    setNewTodo(event.target.value);
+  };
+
+  const handleNewTodoAdd = (event) => {
+    event.preventDefault();
+    if (newTodo.trim() !== "") {
+      setTodos([{ text: newTodo.trim(), completed: false }, ...todos]);
+      setNewTodo("");
+    }
+  };
+
+  const handleTodoDelete = (index) => {
+    const newTodos = [...todos];
+    newTodos.splice(index, 1);
+    setTodos(newTodos);
+  };
+
+  const handleTodoEdit = (index, newText) => {
+    const newTodos = [...todos];
+    newTodos[index].text = newText;
```

Comment:
Foro visibility, this can be put in an external function so to be used in all places.